### PR TITLE
NAS-125612 / None / Simplify mocking of user roles

### PR DESCRIPTION
**Pull request by undsoft**
Also fixes smaller bugs.

Testing: mostly code review.

### CODE REVIEW

**Review by denysbutenko — APPROVED** [review]
Looks good

File: src/app/services/auth/auth.service.ts [hunk 1]
```diff
@@ -172,18 +172,24 @@ export class AuthService {
...
 
-    if (currentRoles.includes(Role.FullAdmin)) {
-      return true;
-    }
+        if (currentRoles.includes(Role.FullAdmin)) {
+          return true;
+        }
 
-    return roles.some((role) => currentRoles.includes(role));
+        return roles.some((role) => currentRoles.includes(role));
```
Comment (RehanY147) [inline src/app/services/auth/auth.service.ts:190]: Is there a scenario where all the roles being checked must exist? Not at least one?

**Review by RehanY147 — APPROVED** [review]

**Review by RehanY147 — CHANGES_REQUESTED** [review]
Shouldn't `hasRole` wait for the user to load? The user loads as soon as the user signs in. So, at most, the wait would be a couple of seconds. As is, the user of the service might have to add a check themselves for the user object emitted by `AuthService` to be truthy.

File: src/app/services/auth/auth.service.ts (hunk 1 above)
Comment (undsoft) [reply src/app/services/auth/auth.service.ts:190]: Not that I'm aware of.

**Comment by undsoft** [comment]
It's a good point, but I don't think `hasRole` has to wait for the user necessarily. The way code is currently structured, user won't get redirected inside before user object is loaded (I think).
I will also refactor `refreshUser` to wait when it is called inside the admin panel.

**Pull request merged**